### ISSUE TITLE
Wrong version number from release 1.1.7-2

### ISSUE THREAD

**Issue by LelieL91** [rustdesk/rustdesk-server#240]
**Describe the bug**
After release 1.1.7-1, every following releases give the wrong package version using ./hbbs -V (or ./hbbr -V either).
I just realise that after making an auto-update bash script that always return me the old version (1.1.7-1) even if latest version is used.

**Describe the environment**
Hardware: Raspberry Pi 3
OS: Raspberry OS (Debian 10 Buster based)
Binary executed in bash shell

**How to Reproduce the bug**
1. Given the previously described environment
2. Create Project Folder and move inside directory
3. wget https://github.com/rustdesk/rustdesk-server/releases/download/1.1.7-4/rustdesk-server-linux-armv7.zip
4. unzip -j rustdesk-server-linux-armv7.zip
5. ./hbbs -V

**Expected behavior**
output: hbbs 1.1.7-4 

**Additional context**
I checked all versions after 1.1.7-1 and all of 3 (1.1.7-2, 1.1.7-3, 1.1.7-4) give as output: 1.1.7-1

**Comment by paspo** [comment]
I think `1.1.7` is the correct *code* version, while `-4` is for packaging purposes only. 
If @rustdesk confirms this, it's better to change the behaviour of the `-V` arg and display only the relevant version.

**Comment by LelieL91** [comment]
> I think `1.1.7` is the correct _code_ version, while `-4` is for packaging purposes only. If @rustdesk confirms this, it's better to change the behaviour of the `-V` arg and display only the relevant version.

Agree about changing the behaviour of the `-V` argument, but by also including the minor version, because it's ok that the correct version is 1.1.7, but the same version got 4 minor updated versions, and since the v1.1.7-1 got the `-1` and new releases tag name got that number changed, I guess it is just a "forgotten edit" error.

**Comment by paspo** [comment]
In fact, @LelieL91 is right: some changes were made (I checked from tag `1.1.7` to `1.1.7-4`) without changing the minor version. [Semantic Versioning](https://semver.org/) is not happy about it.

There are 2 ways out of this:

1. We keep separated the code changes (anything that happens in `src` and `Cargo.*`) from the changes related to any form of packaging. Code changes triggers the version increment, which generates new (arch specific) binaries, that stays the same even if there are some packaging changes. Packaging changes will add a "semantic label" to the version whenever some modification is made to the docker images, the .deb packages, the freebsd port, or the windows build.
2.  We keep it (very) simple and for each change in the real software or in any build we increment the version, using only major, minor and patch for versioning. This will generate a lot of new versions and some of them can be debatable (take for example the change in the build system from ubuntu-18.04 to a more recent one), but will keep versioning simple.

Let's wait for @rustdesk opinion on this.

**Comment by rustdesk** [comment]
Yes, it confused me also, I will update the version number next time once there is Rust code change.

**Comment by pustekuchen91** [comment]
what about switching to calver?

**Comment by paspo** [comment]
> what about switching to calver?

The versioning scheme is not the point here.
The problem is the differentiation between the actual version of the binary itself and the packaging of it.
And the fact that sometimes the binary version and the git tag version are different.

Moving to calver will not resolve any of these issue, at the cost of added confusion due to the versioning change.
Also, calver is more suited to projects where you have a lot of big changes and fast development, in cases where semver will trigger a lot of major revisions. Even if with calver the user is more likely to upgrade (if you're using version 2022, you're feeling old), the target audience of this software is expected to be more conservative.

**Comment by LelieL91** [comment]
Just for info, this "bug" is still found in the latest release v1.1.7-4.
